Issue #3656: Reduce browser icons memory footprint

mauryat opened the pull request:
When the system triggers a low memory event, handle it by clearing the memory cache of browser icons.



### Pull Request checklist
<!-- Before submitting the PR, please address each item -->
- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/master/docs/changelog.md) or does not need one
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### After merge
- [ ] **Milestone**: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.

mauryat commented:
@pocmo I haven't run any pre-push hooks as I couldn't find any defined. Am I missing any?

mauryat commented:
@pocmo I haven't run any pre-push hooks as I couldn't find any defined. Am I missing any?

pocmo commented:
@mauryat Mh, yeah, seems like we never added the configuration to the repository. I run `./gradlew detekt ktlint` as a pre-push hook.

mauryat commented:
> @mauryat Mh, yeah, seems like we never added the configuration to the repository. I run `./gradlew detekt ktlint` as a pre-push hook.

@pocmo thank you! Now it's ready for review. Ran the pre-push hook successfully without any failures.

mauryat commented:
> This looks good to me 👍
> 
> @mauryat Can you squash the two commits and mention this new feature in `docs/changelog.md`? :)

Added changelog and squashed! I thought git allowed squashing while merging, so didn't bother to squash earlier. Will make sure I squash them in the future 😄 !
